managed/utils/auth/token.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (C) 2023 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+// Package auth contains functions to work with auth tokens and headers.
+package auth
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/gogo/status"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+)
+
+// GetTokenFromHeaders returns authorization token if it is found in provided HTTP headers.
+func GetTokenFromHeaders(authHeaders http.Header) string {
+	authHeader := authHeaders.Get("Authorization")
+	switch {
+	case strings.HasPrefix(authHeader, "Bearer"):
+		return strings.TrimSpace(strings.TrimPrefix(authHeader, "Bearer"))
+	case strings.HasPrefix(authHeader, "Basic"):
+		h := strings.TrimPrefix(authHeader, "Basic")
+		t, err := base64.StdEncoding.DecodeString(strings.TrimSpace(h))
+		if err != nil {
```

Comment:
Shouldn't we log this error in `debug` mode?

Reply:
Added.